pkg/prom/instance/configstore/unique.go

```diff
@@ -0,0 +1,35 @@
+package configstore
+
+import (
+	"github.com/grafana/agent/pkg/prom/instance"
+)
+
+// checkUnique validates that cfg is unique from all, ensuring that no two
+// configs share a job_name.
+func checkUnique(all <-chan instance.Config, cfg *instance.Config) error {
+	defer func() {
+		// Drain the channel, which is necessary if we're returning an error.
+		for range all {
+		}
+	}()
+
+	newJobNames := make(map[string]struct{}, len(cfg.ScrapeConfigs))
+	for _, sc := range cfg.ScrapeConfigs {
+		newJobNames[sc.JobName] = struct{}{}
+	}
+
+	for otherConfig := range all {
+		// If the other config is the one we're validating, skip it.
+		if otherConfig.Name == cfg.Name {
+			continue
+		}
+
```

Comment:
Not in love with this, but I cant think of a better way to do it with the default set of golang collections.

Reply:
I don't like the uniqueness checking at all honestly. We have #398 in the backlog to try to avoid this altogether